content/en/blog/_posts/2022-11-11-retroactive-default-storage-class.md

```diff
@@ -0,0 +1,131 @@
+---
+layout: blog
+title: "Retroactive default StorageClass"
+date: 2022-11-11
+slug: retroactive-default-storage-class
+---
+
+**Author:** Roman Bednář (Red Hat)
+
+In Kubernetes 1.25 a new Alpha feature was introduced to improve the way default StorageClass is assigned to a PersistentVolumeClaim (PVC). It is no longer required to create a default StorageClass first and PVC second in order to assign the class and any PVCs without StorageClass can be updated later. The feature will be Beta in Kubernetes 1.26.
+
+See [Official Kubernetes documentation](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#retroactive-default-storageclass-assignment) for more details.
+
+## Why did StorageClass assignment need improvements
+
+When talking about PVCs and default StorageClasses you might think "It already works, when I create PVC it gets the default class just fine!" and that is correct because there already is a feature that causes admission controller to assign a class to **new** PVCs which is documented [here](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#defaultstorageclass).
+
+But what if there was no default StorageClass at the time of PVC creation? Users end up with a PVC that will never get a class and so no storage will be provisioned, the PVC is somewhat "stuck" at this point. From high level point of view there were two main use cases which might result in "stuck" PVCs and cause problems later down the road. Let's take a closer look at each of them.
+
+### Changing default StorageClass did not have a good solution
+
+   There are two options admins had when they wanted to change default StorageClass:
+   1. Creating the new StorageClass as default before removing the old one which results in having two defaults for a short period of time. If a user creates a PersistentVolumeClaim at this point with storageClassName set to `nil` (which means default StorageClass) Kubernetes 1.26+ will choose the newest default StorageClass and assign it to this PVC.
```

Comment:
1. Creating the new StorageClass as default before removing the old one which results in having two defaults for a short period of time. If a user creates a PersistentVolumeClaim at this point with storageClassName set to `null` (which means default StorageClass), Kubernetes 1.26+ will choose the newest default StorageClass and assign it to this PVC.

Reply:
1. Do we want to mix `nil` and `null` terms? Won't that be confusing?

2. Above you mentioned avoiding statements like *"Kubernetes 1.26+ will..."* - while in this case this behavior is already [merged](https://github.com/kubernetes/kubernetes/pull/110559). I don't even have to mention the version here probably as it does not add any value - removing.